content/post/2018-06-15-serial_number.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Provisioning Nerves Devices
+subtitle: Life after nerves.local
+date: 2018-06-15
+author: Frank Hunleth
+draft: false
+tags: ["nerves", "provisioning"]
+---
+
+When you're starting out with Nerves, you may have connected to your first
+projects over the network using `nerves.local`. Projects like
+[nerves_init_gadget](nerves_init_gadget) make this easy and when you're
+starting out, it's really convenient. Don't know the IP address that your
+device was assigned? Try `nerves.local` and you're good to go.
+
+And then you add a second device to your network. `nerves.local` isn't looking
+so convenient any more.
+
+<!--more-->
+
+Suffice it to say that there are a number of ways of solving this problem. This
+post shows how to provision names or numbers to devices. You'll likely need to
+provision additional information to devices and that can be done in a similar
+way.
+
+The first step is to decide on a way of identifying devices. Nearly every
+Nerves System has a way of finding a unique identifier for a board by default.
+The default is to use that number to create a unique hostname and you can see
+it by running `:inet.gethostname()`. It will be something like `nerves-1234`.
+(This hostname is only local unless something registers it in the DNS.
+`nerves.local` is a mDNS name.) The `erlinit.config` file specifies how to
+create the hostname. For the [Raspberry Pi 3](rpi3_erlinit) look for the
+following:
```

Comment:
> For the [Raspberry Pi 3](rpi3_erlinit) look for the following:

I am thinking

In the Raspberry Pi 3 [erlinit.config](rpi3_erlinit) look for the following lines: